src/Gui/ExpressionCompleter.cpp

```diff
@@ -399,36 +679,75 @@ QStringList ExpressionCompleter::splitPath ( const QString & input ) const
         return l;
 
     int retry = 0;
-    std::string trim;
+    std::string lastElem;  // used to recover in case of parse failure after ".". 
+    std::string trim;      // used to delete ._self added for another recovery path 
     while(1) {
         try {
+            // this will not work for incomplete Tokens at the end
+            // "Sketch." will fail to parse and complete.
+
             App::ObjectIdentifier p = ObjectIdentifier::parse(
```

Comment:
This code will collapse any global references to the current object to the sub path.

It causes issues when you have objects or props with the same name as documents, or other objects. 

Example `Constraints` object in Assembly4 blocks completion for a local `Constraints`. Looking at solutions.